src/Symfony/Component/Messenger/Tests/Transport/Doctrine/DoctrineIntegrationTest.php

```diff
@@ -21,29 +21,24 @@
  */
 class DoctrineIntegrationTest extends TestCase
 {
+    /** @var \Doctrine\DBAL\Connection */
     private $driverConnection;
+    /** @var Connection */
     private $connection;
 
-    /**
-     * @after
-     */
-    public function cleanup()
-    {
-        @unlink(sys_get_temp_dir().'/symfony.messenger.sqlite');
-    }
-
-    /**
-     * @before
-     */
-    public function createConnection()
+    protected function setUp(): void
     {
         $dsn = getenv('MESSENGER_DOCTRINE_DSN') ?: 'sqlite:///'.sys_get_temp_dir().'/symfony.messenger.sqlite';
         $this->driverConnection = DriverManager::getConnection(['url' => $dsn]);
         $this->connection = new Connection([], $this->driverConnection);
-        // call send to auto-setup the table
-        $this->connection->setup();
-        // ensure the table is clean for tests
-        $this->driverConnection->exec('DELETE FROM messenger_messages');
+    }
+
+    protected function tearDown(): void
+    {
+        $this->driverConnection->close();
+        if (file_exists(sys_get_temp_dir().'/symfony.messenger.sqlite')) {
+            unlink(sys_get_temp_dir().'/symfony.messenger.sqlite');
+        }
     }
```

Comment:
These changes were related to a Windows only issue with the SQLite database file. Not sure why that issue started with this PR, but this new way of ensuring a clean database looks good.

Reply:
for reference, the previous `@unlink` was failing with `Resource temporarily unavailable`. Closing the connection is the fix.

Reply:
`@unlink(sys_get_temp_dir().'/symfony.messenger.sqlite');` should be enough, right?

Reply:
yes, but the `@` was hiding the previous error causing the tests to fail for unknown reasons, (which took me a hour to debug pushing `var_dumps` to appveyor) which is why i have removed the `@`.

Reply:
```php
        if (file_exists($file = sys_get_temp_dir().'/symfony.messenger.sqlite')) {
            unlink($file);
        }
```

Reply:
done